### ISSUE TITLE
Documentation: DeferredSemaphore needs to document its __init__ arguments

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4443]
|<img alt="jrennie's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jrennie reported|
|-|-|
|Trac ID|trac#4443|
|Type|enhancement|
|Created|2010-05-19 14:08:46Z|
|Branch|https://github.com/twisted/twisted/tree/deferredsemaphore-tokens-doc-4443|

One has to look at the source code to infer that "tokens" is an integer which must be >= 1.  Also, the __init__ method really ought to perform the corresponding safety check:

if tokens < 1 or round(tokens) != tokens:
    raise Exception("...")

<details><summary>Searchable metadata</summary>

```
trac-id__4443 4443
type__enhancement enhancement
reporter__jrennie jrennie
priority__normal normal
milestone__ 
branch__branches_deferredsemaphore_tokens_doc_4443 branches/deferredsemaphore-tokens-doc-4443
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1274278126000000 1274278126000000
changetime__1274715928000000 1274715928000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [28973]) Branching to 'deferredsemaphore-tokens-doc-4443'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [28974]) Document tokens and limit and enforce tokens to be positive

refs [#4443](https://github.com/twisted/twisted/issues/4443)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

I'm a little unsure about the `round(tokens) != tokens` suggestion.  But I implemented the rest.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

(minor edit to [correct grammar](http://www.angryflower.com/bobsqu.gif) in the summary)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @exarkun|
|-|-|

Looks good to me, merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [29030]) Merge deferredsemaphore-tokens-doc-4443 

Author: exarkun
Reviewer: glyph
Fixes: [#4443](https://github.com/twisted/twisted/issues/4443)

Document the `tokens` and `limit` attributes of `DeferredSemaphore`, and reject
invalid non-positive values for `token` in `DeferredSemaphore`'s initializer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
